packages/ai-jsx/src/lib/tts/tts.tsx

```diff
@@ -1,6 +1,207 @@
 'use client';
 import { split } from 'sentence-splitter';
 
+const AUDIO_WORKLET_SRC = `
+class OutputProcessor extends AudioWorkletProcessor {
+  constructor() {
+    super();
+    this.msgs = {};
+    this.seqNum = 0;
+    this.bufPos = 0;
+    this.port.onmessage = (e) => {
+      this.msgs[e.data.seqNum] = e.data;
+      //console.debug('buf added, seq num = ' + e.data.seqNum + ' queue len=' + this.queue.length.toString());
+    } 
+  }
+  process(inputs, outputs) {
+    const output = outputs[0];
+    const outLeft = output[0];
+    const outRight = output[1];
+    const outLen = outLeft.length;
+    this.copy(outLeft, outRight, outLen);
+    return true;
+  }
+  copy(outLeft, outRight, outLen) {  
+    let outPos = 0;
+    while (outPos < outLen) {
+      const buf = this.buf();
+      if (!buf) {
+        break;
+      }
+      const bufLeft = buf[0];
+      const bufRight = buf[1] ?? bufLeft;
+      const bufLen = bufLeft.length;      
+      const len = Math.min(bufLen - this.bufPos, outLen - outPos);
+      outLeft.set(bufLeft.subarray(this.bufPos, this.bufPos + len), outPos);
+      outRight?.set(bufRight.subarray(this.bufPos, this.bufPos + len), outPos);
+      outPos += len;
+      this.bufPos += len;      
+    }
+    if (outPos < outLen) {
+      //console.warn('buf exhausted, out pos=' + outPos.toString());
+      outLeft.fill(0, outPos);
+      outRight?.fill(0, outPos);
+      outPos = outLen;
+    }
+  }
+  buf() {
+    const msg = this.msgs[this.seqNum];
+    if (!msg) {
+      return null;
+    }
+    if (this.bufPos == msg.channelData[0].length) {
+      this.port.postMessage({seqNum: msg.seqNum});
+      delete this.msgs[msg.seqNum];
+      //console.debug('buf consumed, seq num=' + msg.seqNum.toString() + ' queue len=' + this.queue.length.toString());
+      this.seqNum++;
+      this.bufPos = 0;
+      return this.buf();
+    }
+    return msg.channelData;
+  }
+}
+
+registerProcessor("output-processor", OutputProcessor);
+`;
+
+const AUDIO_MPEG_MIME_TYPE = 'audio/mpeg';
+const AUDIO_PCM_MIME_TYPE = 'audio/pcm';
+const AUDIO_L16_MIME_TYPE = 'audio/L16';
+
+class AudioChunk {
+  mimeType: string;
+  sampleRate: number;
+  constructor(contentType: string, public buffer: ArrayBuffer) {
+    const pieces = contentType.split(';');
+    this.mimeType = pieces[0];
+    const sampleRatePiece = pieces.find((piece) => piece.trim().startsWith('rate='));
+    this.sampleRate = sampleRatePiece ? parseInt(sampleRatePiece.split('=')[1]) : 0;
+  }
+  get isPcm() {
+    return this.mimeType == AUDIO_PCM_MIME_TYPE || this.mimeType == AUDIO_L16_MIME_TYPE;
+  }
+}
+
+class AudioMessage {
+  constructor(public seqNum: number, public channelData: Float32Array[]) {}
+}
```

Comment:
This can also just be an interface.

Reply:
OOC, why use an interface here? And how would I pass this to `postMessage` since I can no longer do `new AudioMessage()`?

Reply:
```ts
// as a class
const audioMessage = new AudioMessage(seqNum, channelData);

// as an interface
const audioMessage = { seqNum, channelData }
```

Reply:
You can't just pass that object to `postMessage`?

Reply:
Sure, but then it's no longer typed, right? I.e., what's the point of the interface definition if I'm just using {}?

Reply:
You generally type it at the place where you need it to have a type, such as a function param. This maximizes flexibility. 

But if you want to ensure it has a type at creation time, you can do:

```
const a: TypeName = { key: val }
```

Reply:
Types are, of course, erased at runtime.